avoidance/include/avoidance/common.h

```diff
@@ -214,6 +215,16 @@ void transformVelocityToTrajectory(mavros_msgs::Trajectory& obst_avoid,
 * @param      point, setpoint to be filled with NAN
 **/
 void fillUnusedTrajectoryPoint(mavros_msgs::PositionTarget& point);
-}
+
+/**
+* @brief           This is a refactored version of the PCL library function to
+*                  remove NAN values from the point cloud and compute the FOV
+* @note            It operates in-place and iterates through the cloud once
+* @param[in, out]  cloud The point cloud to be filtered in the camera frame
+* @param[in, out]  fov of the camera
+**/
+void removeNaNFromPointCloud(pcl::PointCloud<pcl::PointXYZ>& cloud, FOV& fov);
```

Comment:
This also calculates the FOV, so I'd stick that in the name somewhere too

Reply:
agreed